platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -357,6 +356,8 @@ public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float ve
         return false;
       }
 
+      cameraChangeDispatcher.onCameraMoveStarted(REASON_API_GESTURE);
```

Comment:
I think this could be moved below `transform.cancelTransitions();` so that `onCameraMoveCanceled()` is called before `onCameraMoveStarted()`